typings/index.d.ts

```diff
@@ -241,10 +241,19 @@ export class Help {
   padWidth(cmd: Command, helper: Help): number;
 
   /**
-   * Wrap the given string to width characters per line, with lines after the first indented.
-   * Do not wrap if insufficient room for wrapping (minColumnWidth), or string is manually formatted.
+   * Merge left text column defined by first `leadingStrLength` characters of `str` with right text column defined by remaining characters, wrapping the output to `width - 1` characters per line.
+   *
+   * Do not wrap if right column text is manually formatted.
+   *
+   * Lines containing left column are indented by `globalIndent - Math.min(0, fullIndent)` and all new lines due to overflow by `fullIndent` if it is positive and does not cause text display width to be too narrow, where `fullIndent = globalIndent + leadingStrWidth + columnGap + overflowShift` with `leadingStrWidth` being the computed width of the left column. `leadingStrWidth` and `columnGap` are omitted from the computation if right column text is manually formatted.
+   *
+   * `leadingStrLength`, `overflowShift`, `globalIndent` and `columnGap` all default to 0.
+   *
+   * Text display width is considered too narrow when it is less than `minWidthGuideline` which defaults to 40.
+   *
+   * Unless `preformatted` is specified explicitly, right column text is considered manually formatted if it includes a line break followed by a whitespace.
    */
-  wrap(str: string, width: number, indent: number, minColumnWidth?: number): string;
+  wrap(str: string, width: number, leadingStrLength?: number, minWidthGuideline?: number, overflowShift?: number, globalIndent?: number, columnGap?: number, preformatted?: boolean): string;
```

Comment:
Yeah, all the indents simply listed one after the other can be quite confusing, so I agree with you, except maybe `preformatted` should be additionally available as a separate argument, just like `minWidthGuideline` is now. Both are special because they are in essence universal settings that are supposed to have the same value in each call in any reasonable application, so ideally, they should only be used directly when overwriting `wrap()` in a subclass. Because of that similarity and since `minWidthGuideline` is already available as a separate argument, I think `preformatted` should be, too. We could also make them both properties of the `Help` class instead, so that the values could be configured with `Command.configureHelp()`.

Anyway, I got stuck trying to write JSDocs for the overloads with an `options` parameter. Apparently, VS Code does not understand overloaded functions well, so it led me to create this issue: https://github.com/microsoft/vscode/issues/187663.

Reply:
[fbddb6b](https://github.com/tj/commander.js/pull/1904/commits/fbddb6bbc9498af7f6f9e15e2a9f1e450f7535a0)